### PR TITLE
bpo-38165: Fix tempfile.py

### PR DESCRIPTION
**Pull request by evgeniysolntsev**
freezes when loading a program on imports because user rights are insufficient
python installed on c:/ and requires administrator rights
when run program from not root user

### CODE REVIEW

**Review by eduardo-elizondo — CHANGES_REQUESTED** [review]
All the builds are failing. Please look at the failing builds and re-submit your change

**Review by aeros — COMMENTED** [review]
Thanks for the PR @evgeniymischuk. Please adjust the title of the PR according the standard conventions. The format is: ``bpo-<issue_number>: <summary_of_PR>``. This allows the PR to be directly linked to the bpo issue. If you haven't already, I would highly recommend reading through the [devguide's quick reference page](https://devguide.python.org/#quick-reference). 

Travis CI is currently failing because of a error in ``test.test_tempfile.TestMkstempInner``, specifically for the tests ``test_collision_with_existing_directory`` and ``test_collision_with_existing_file ``. Both failures are occurring during the handling of ``FileNotExistsError``. Here's the relevant traceback, for debugging purposes:
image

I didn't include the other one since they both contained the same exact traceback. Essentially, when the ``FileExistsError`` is being raised, there is no suitable replacement name found for the temporary file created during the tests. I would recommend trying to address this issue first.

Before you update this PR with a new batch of commits, you should [run the tests locally](https://devguide.python.org/runtests/#running) to ensure everything is passing. The CI can still potentially fail if the tests are not passing in other environments, but this will help to rule out the majority of issues.

**Comment by aeros167** [comment]
For now, I am adding a ``DO-NOT-MERGE`` label since there are issues to resolve with the PR. Once the problems have been addressed, either someone else can remove the label, or you can @mention me to take a look at the PR again.

File: Lib/tempfile.py [hunk 1]
```diff
@@ -242,26 +242,16 @@ def _mkstemp_inner(dir, pre, suf, flags, output_type):
     if output_type is bytes:
         names = map(_os.fsencode, names)
 
-    for seq in range(TMP_MAX):
```
Comment (aeros) [inline Lib/tempfile.py:245]: Why is this being removed? This block is supposed to continue trying to create the file until the maximum number of tries are reached, ``TMP_MAX``.

File: Lib/tempfile.py [hunk 2]
```diff
@@ -242,26 +242,16 @@ def _mkstemp_inner(dir, pre, suf, flags, output_type):
...
-                continue
-            else:
-                raise
-        return (fd, _os.path.abspath(file))
+    name = next(names)
+    file = _os.path.join(dir, pre + name + suf)
+    try:
+        fd = _os.open(file, flags, 0o600)
+    except FileExistsError:
+        raise FileExistsError(_errno.EEXIST, "No usable temporary file name found")
```
Comment (aeros) [inline Lib/tempfile.py:250]: When this exception occurs, this block is supposed try to *handle* the exception, not just raise it.

File: Lib/tempfile.py [hunk 3]
```diff
@@ -242,26 +242,16 @@ def _mkstemp_inner(dir, pre, suf, flags, output_type):
...
+    except PermissionError:
+        raise PermissionError(_errno.EACCES, "No permission to create temp file")
```
Comment (aeros) [inline Lib/tempfile.py:252]: See my [other comment](https://github.com/python/cpython/pull/16126/files#r324393420).

**Comment by matrixise** [comment]
Hi @aeros167 

Thank you for your review and your time, but I think we can close this PR and the associated issues.

**Comment by aeros167** [comment]
> Hi @aeros167
Thank you for your review and your time, but I think we can close this PR and the associated issues.

No problem, I think they could be closed as well.

**Pull request closed without merging**